package.json

```
@@ -0,0 +1,47 @@
+{
+  "name": "mesto_webpack",
+  "homepage": "https://github.com/Mlenizyaka/Mesto_Webpack#readme",
+  "version": "0.0.1",
+  "description": "[Демо](https://mlenizyaka.github.io/Mesto_Webpack.github.io/)",
+  "main": "index.js",
+  "scripts": {
+    "build": "cross-env NODE_ENV=production rimraf dist && webpack --mode production",
```

🦀 Понятно, что этот код был взят как есть из практики, поэтому не отношу его ни к одной из секций. 🦀 

Если залогировать значение `process.env.NODE_ENV` в `webpack.config.js` мы получим следующее:
```bash
$ npm run build

> mesto_webpack@0.0.1 build ~/Mesto_Webpack
> cross-env NODE_ENV=production rimraf dist && webpack --mode production

{ NODE_ENV: undefined }
```

Это связано с тем, что при выполнении команды `npm run build` будет выполнено 2-е команды последовательно (они соединены через `&&`):
1. `cross-env NODE_ENV=production rimraf dist`.
2. `webpack --mode production`

Видно, что ENV переменная не передается 2-й команде соответственно в коде `process.env.NODE_ENV` равен `undefined`. 

Чтобы работало, нужно `cross-env NODE_ENV=production` перенести во вторую команду:
```suggestion
    "build": "rimraf dist && cross-env NODE_ENV=production webpack --mode production",
```

Результаты выполнения:
```bash
$ npm run build

> mesto_webpack@0.0.1 build ~/Mesto_Webpack
> cross-env NODE_ENV=production rimraf dist && webpack --mode production

{ NODE_ENV: 'production' }
```